### PR TITLE
RUBY-2808 Add a CMAP test that verifies the background thread hands over connections to threads doing checkout

### PR DESCRIPTION
**Pull request by neilshweky**
Look at #2641 when this is done

Still failing tests related to this. I cannot reproduce them.

### CODE REVIEW

File: spec/runners/cmap/verifier.rb [hunk 1]
```diff
@@ -41,7 +41,9 @@ def verify_hashes(actual, expected)
           end
         end
 
-        expect(actual_modified).to eq(expected)
+        expected.each do |k, v|
```
Comment (Neilshweky) [inline spec/runners/cmap/verifier.rb:44]: The spec describes the match pseudocode [here](https://github.com/mongodb/specifications/blob/master/source/connection-monitoring-and-pooling/tests/README.rst#L158):
```
 For every key/value in expected
        Assert that actual[key] MATCHES value
```

Comment (p-mongo) [reply spec/runners/cmap/verifier.rb:44]: If it's in the spec...

**Comment by p-mongo** [comment]
I think this is going to conflict with my PR so I'd rather wait here.

**Comment by Neilshweky** [comment]
okay sure @p-mongo I don't mind waiting on this one.

**Comment by Neilshweky** [comment]
These tests fail on JRuby in CI. I cannot reproduce locally.

**Pull request closed without merging**